File: docs/guides/csharp/src/searchWithRuleContexts.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SearchWithRuleContexts
{
  private static string GetPlatformTag()
  {
    return ""; // Implement your logic here
  }

  async Task Main(string[] args)
  {
    var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));
    var platformTag = GetPlatformTag();
    var searchParams = new SearchParams(
      new SearchParamsObject { Query = "<YOUR_SEARCH_QUERY>", RuleContexts = [platformTag] }
    );

    await client.SearchSingleIndexAsync<Hit>("<YOUR_INDEX_NAME>", searchParams);
  }
}

File: docs/guides/go/src/searchWithRuleContexts.go
package main

import (
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func getPlatformTag() (string, error) {
	return "", nil // Implement your logic here
}

func searchWithRuleContexts() {
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	platformTag, err := getPlatformTag()
	if err != nil {
		panic(err)
	}

	searchParams := search.SearchParamsObjectAsSearchParams(
		search.NewSearchParamsObject().
			SetQuery("<YOUR_SEARCH_QUERY>").
			SetRuleContexts([]string{platformTag}),
	)

	_, err = client.SearchSingleIndex(client.NewApiSearchSingleIndexRequest(
		"<YOUR_INDEX_NAME>").WithSearchParams(searchParams))
	if err != nil {
		panic(err)
	}
}
